paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerPreparesGrindstoneCraftScriptEvent.java

```diff
@@ -0,0 +1,115 @@
+package com.denizenscript.denizen.paper.events;
+
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizen.objects.*;
+import com.denizenscript.denizen.utilities.implementation.BukkitScriptEntryData;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.scripts.ScriptEntryData;
+import com.denizenscript.denizencore.utilities.CoreUtilities;
+import com.destroystokyo.paper.event.inventory.PrepareResultEvent;
+import org.bukkit.entity.HumanEntity;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.inventory.GrindstoneInventory;
+import org.bukkit.inventory.Inventory;
+
+public class PlayerPreparesGrindstoneCraftScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // player prepares grindstone craft <item>
+    //
+    // @Group Player
+    //
+    // @Location true
+    //
+    // @Plugin Paper
+    //
+    // @Triggers when a player prepares to grind an item.
+    //
+    // @Context
+    // <context.inventory> returns the InventoryTag of the grindstone inventory.
+    // <context.result> returns the ItemTag to be crafted.
+    //
+    // @Determine
+    // "RESULT:" + ItemTag to change the item that is crafted.
+    //
+    // @Player Always.
+    //
+    // @Warning The player doing the grinding is estimated and may be inaccurate.
+    //
+    // @Examples
+    // This example removes also the cursed binding_curse enchantment from the item.
+    // on player prepares grindstone craft item:
+    // - determine result:<context.result.with[remove_enchantments=binding_curse]>
+    //
+    // -->
+
+
+    public PlayerPreparesGrindstoneCraftScriptEvent() {
+        registerCouldMatcher("player prepares grindstone craft <item>");
+    }
+
+    public PrepareResultEvent event;
+    public ItemTag item;
+    public Inventory inventory;

```

Comment:
Is there any reason to have these fields? `item` is only used once so can construct the `ItemTag` there, and `inventory` is just `event.getInventory()` so can probably use that directly

Reply:
Removed fields.